hoomd/md/ManifoldSphere.h

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2009-2021 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+
+// Maintainer: pschoenhoefer
+
+#ifndef __MANIFOLD_CLASS_SPHERE_H__
+#define __MANIFOLD_CLASS_SPHERE_H__
+
+#include "hoomd/HOOMDMath.h"
+#include "hoomd/BoxDim.h"
+#include <pybind11/pybind11.h>
+
+/*! \file ManifoldSphere.h
+    \brief Defines the manifold class for the Sphere surface
+*/
+
+// need to declare these class methods with __device__ qualifiers when building in nvcc
+// DEVICE is __host__ __device__ when included in nvcc and blank when included into the host compiler
+#ifdef __HIPCC__
+#define DEVICE __device__
```

Comment:
Shouldn't this definition be HOSTDEVICE __host__ __device__? You still want to be able to run simulations on CPU in a GPU build.

Reply:
the markdown is changing the dunders in the above comment to bold, but you know what I mean.

Reply:
No, it doesn't need to be. The reason for this is related to @vyasr's questions about putting the host code in a .cc file. 

In HOOMD, the CPU compilation units are all in `.cc` files and compiled by the host compiler natively. We only use `.cu` files for GPU kernels and their driver functions. `nvcc` uses the `__host__` to determine which functions to include in its own host/device split of the code. 

This model exists for two reasons. 1) In the past, `nvcc` has chocked on compiling C++ code like pybind11. Thus, we separate that code all out into code compiled only by the host compiler. 2) If we were to rely on `nvcc` to split the host and device code for us, then we would need to compile all of HOOMD with `nvcc` even for CPU only builds. This requirement would be too much burden for many HOOMD users, packaging systems (i.e. conda-forge), and would make it impossible (or very hard) to build HOOMD on Mac where `nvcc` is no longer supported as far as I know.